lib/Agrammon/UI/CommandLine.pm6

```diff
@@ -41,20 +43,30 @@ multi sub MAIN('web', ExistingFile $cfg-filename, ExistingFile $model-filename,
 #| Run the model
 multi sub MAIN('run', ExistingFile $filename, ExistingFile $input, Str $tech-file?,
                SupportedLanguage :$language = 'de', Str :$prints = 'All', Str :$variants = 'SHL',
-               Bool :$csv, Bool :$include-filters, Int :$batch=1, Int :$degree=4, Int :$max-runs
+               Bool :$csv, Bool :$include-filters, Int :$batch=1, Int :$degree=4, Int :$max-runs,
+               OutputFormat :$format = 'txt'
               ) is export {
     my %results = run $filename.IO, $input.IO, $tech-file, $variants, $language, $prints, $csv, $include-filters,
             $batch, $degree, $max-runs;
-    say "##  Model: $filename";
-    say "##  Variants: $variants";
-    for %results.keys -> $simulation {
-        say "### Simulation $simulation";
-        say "##  Print filter: $prints";
-        for %results{$simulation}.keys.sort -> $dataset {
-            say "#   Dataset $dataset";
-            say %results{$simulation}{$dataset};
+    my $output;
+    if $format eq 'json' {
+        $output = to-json %results;
+    }
+    else {
+        my @output;
+        @output.push("##  Model: $filename");
+        @output.push("##  Variants: $variants");
+        for %results.keys -> $simulation {
```

Comment:
Could be `for %results.kv -> $simulation, %sim-results {` to save a hash indexing later

Reply:
Thanks. Just realized, that it is not quite ready yet.